### PR TITLE
Fixed lambda_function to prevent errors during simultaneous deployments

### LINKED ISSUE #46: [BUG]
**Describe the bug**
When we have multiple pipelines running at the same time (e.g., during releases), the lambda runs in parallel and we get exceptions thrown when the disassociation does not fine the alias or the old index.

Background: As a part of website and docsite deployment, a script creates a payload of the content of the websites for a custom search. This payload is dumped into OpenSearch using a lambda function written in python. The lambda function dumps the payload into an index and then sets an alias to the new index. It then disassociates the alias from the previous index.

### PR DESCRIPTION
**Pull request by saimedhi**
### Description
Fixed lambda_function to prevent errors during simultaneous deployments

### Issues Resolved
closes #46 
### Check List
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/project-website-search/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: cdk/opensearch-website-search/lambdas/indexer-lambda/doc-indexer.py [hunk 1]
```diff
@@ -250,13 +250,15 @@ def do_indexing(os_client, user_params):
                                  request_timeout=20)
     print("Existing alias - Bulk response: ", bulk_response)
     print(SECTION_SEPARATOR)
-
+    
+    indices_to_remove = [index for index in os_client.indices.get_alias(name=INDEX_ALIAS)]
```
Comment (harshavamsi) [inline cdk/opensearch-website-search/lambdas/indexer-lambda/doc-indexer.py:254]: Could you maybe talk about why this change works?

File: cdk/opensearch-website-search/lambdas/indexer-lambda/doc-indexer.py [hunk 2]
```diff
@@ -250,13 +250,15 @@ def do_indexing(os_client, user_params):
...
+    
+    actions=[{"remove": {"index": index, "alias": INDEX_ALIAS}} for index in indices_to_remove]
+    actions.append({"add": {"index": new_index, "alias": INDEX_ALIAS}})
+    
     alias_status = os_client.indices.update_aliases(
       {
-        "actions": [
-          {"remove": {"index": old_index, "alias": INDEX_ALIAS}},
```
Comment (saimedhi) [inline cdk/opensearch-website-search/lambdas/indexer-lambda/doc-indexer.py:257]: Previously, we tried to remove old index. This will cause issues with parallel deployments. When we try to remove old index, it may have already been deleted by another parallel thread. So, here ideal solution will be to remove all the indices that are present and add new index. By this we can always get number of indices mapped to index aliases as one or zero(at initial state).

**Review by AMoo-Miki — CHANGES_REQUESTED** [review]
The information in these indices come from various sources. When parallel requests land, we do want both of them. The current code recognizes that it has lost the reference to the old index and decides to fail rather that have invalid data. By not removing the old ones, this PR would only mask the failure but result in invalid data. 

What we need is for the process to recognize that there is another one running and wait for its completion before it continues.
